CMakeLists.txt

```
@@ -56,27 +65,34 @@ endif()
 # Library
 option(BUILD_SHARED_LIBS "Build shared libraries" ON)
 if(NOT DEFINED CMAKE_INSTALL_RPATH_USE_LINK_PATH)
-  set(CMAKE_INSTALL_RPATH_USE_LINK_PATH ON CACHE STRING
-      "Inform installed binaries of external library rpaths")
+  # Inform installed binaries of external library rpaths
+  set(CMAKE_INSTALL_RPATH_USE_LINK_PATH ON)
 endif()
-if(BUILD_SHARED_LIBS)
-  set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_FULL_LIBDIR}" CACHE STRING
-      "Inform installed binaries of internal library rpaths")
-endif()
-if(APPLE)
-  option(CMAKE_MACOSX_RPATH "Support @rpath in install targets" ON)
+if(BUILD_SHARED_LIBS AND NOT DEFINED CMAKE_INSTALL_RPATH)
+  # Inform installed binaries of internal library rpaths
+  set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_FULL_LIBDIR}")
 endif()
 
 # Build flags
 option(CELERITAS_DEBUG "Enable runtime assertions" ON)
-if(NOT CMAKE_BUILD_TYPE AND (CMAKE_GENERATOR STREQUAL "Ninja"
-    OR CMAKE_GENERATOR STREQUAL "Unix Makefiles"))
-  set(CMAKE_BUILD_TYPE "Debug" CACHE STRING
-    "Build type set to default by Celeritas CMakeLists" FORCE)
+if(DEFINED CMAKE_BUILD_TYPE AND NOT CMAKE_BUILD_TYPE)
```

Why is this not:
```suggestion
if(NOT DEFINED CMAKE_BUILD_TYPE OR NOT CMAKE_BUILD_TYPE)
```
or simply
```suggestion
if(NOT CMAKE_BUILD_TYPE)
```
?

---

`CMAKE_BUILD_TYPE` isn't defined for all generator types: some generators support multiple build types simultaneously, in which case the variable isn't defined and shouldn't be set.